src/StudioComponents/Loading.lua

```diff
@@ -0,0 +1,129 @@
+-- Written by @boatbomber
+
+local Plugin = script:FindFirstAncestorWhichIsA("Plugin")
+local Fusion = require(Plugin:FindFirstChild("Fusion", true))
+
+local StudioComponents = script.Parent
+local StudioComponentsUtil = StudioComponents:FindFirstChild("Util")
+
+local getState = require(StudioComponentsUtil.getState)
+local themeProvider = require(StudioComponentsUtil.themeProvider)
+local constants = require(StudioComponentsUtil.constants)
+local unwrap = require(StudioComponentsUtil.unwrap)
+local types = require(StudioComponentsUtil.types)
+local stripProps = require(StudioComponentsUtil.stripProps)
+
+
+local Computed = Fusion.Computed
+local Hydrate = Fusion.Hydrate
+local Value = Fusion.Value
+local New = Fusion.New
+local Spring = Fusion.Spring
+local Observer = Fusion.Observer
+local Children = Fusion.Children
+
+local COMPONENT_ONLY_PROPERTIES = {
+	"Enabled",
+}
+
+type LoadingProperties = {
+	Enabled: (boolean | types.StateObject<boolean>)?,
+	[any]: any,
+}
+
+local cos = math.cos
+local clock = os.clock
+local pi4 = 12.566370614359172 --4*pi
+
+return function(props: LoadingProperties): Frame
+	local time = Value(0)
+
+	local function startMotion()
+		local startTime = clock()
+		while unwrap(props.Enabled) do
+			time:set(clock()-startTime)
+			task.wait(1/25) -- Springs will smooth out the motion so we needn't bother with high refresh rate here
+		end
+	end
+
+	task.defer(startMotion)
+	Observer(props.Enabled):onChange(function()
```

Comment:
So in the type LoadingProperties `Enabled` is said to be allowed to be a pure boolean, but you can't use an Observer on a boolean. To resolve this with ease, I'd personally just define this at the top before the `time` variable:
```lua
local isEnabled = getState(props.Enabled, true)
```
Since that would be consistent with the rest of the components. And then all you'd need to do is change `props.Enabled` to `isEnabled`

Reply:
I also just noticed, but the `:onChange()` connection to the observer is never cleaned up.